python/tvm/script/printer/doc.py

```diff
@@ -29,6 +29,21 @@
 class Doc(Object):
     """Base class of all Docs"""
 
+    @property
+    def source_paths(self) -> Sequence[ObjectPath]:
+        """
+        The list of object paths of the source IR node.
+
+        This is used to trace back to the IR node position where
+        this Doc is generated, in order to position the diagnostic
+        message.
+        """
+        return self.__getattr__("source_paths")  # pylint: disable=unnecessary-dunder-call
```

Comment:
A really dumb question: isn't this attribute accessible already with `self.source_paths`?

Reply:
Without this property definition, `source_paths` is accessible but cannot be mutated from Python, because the Python `Object` base class only has `__getattr__`, and doesn't have `__setattr__`.